packages/frontend/src/fetcher/on-chain/index.ts

```diff
@@ -108,62 +112,70 @@ class Fetcher implements IPartialFetcher {
-                const logs = await provider.getLogs({
-                    address: realityContract.address,
+                // FIXME: check if this implementation still works properly
+                const newAnwersEventLogs = await nodeClient.getLogs({
+                    address: realityV3Address as `0x${string}`,
+                    event: parseAbiItem(
+                        "event LogNewAnswer(bytes32,bytes32,bytes32,address,uint256,uint256,bool)"
+                    ),
                     fromBlock,
                     toBlock,
-                    topics: [
-                        [NEW_ANSWER_LOG_TOPIC, NEW_QUESTION_LOG_TOPIC],
-                        questionId,
-                    ],
                 });
+                const newQuestionEventLogs = await nodeClient.getLogs({
+                    address: realityV3Address as `0x${string}`,
+                    event: parseAbiItem(
+                        "event LogNewQuestion(bytes32,address,uint256,string,bytes32,address,uint32,uint32,uint256,uint256)"
+                    ),
+                    fromBlock,
+                    toBlock,
+                });
+                newQuestionEventLogs;
+
+                const logs = [...newAnwersEventLogs, ...newQuestionEventLogs];
                 let shouldBreak = false;
                 for (const log of logs) {
+                    // FIXME: is this condition still valid
                     if (log.topics[0] === NEW_ANSWER_LOG_TOPIC) {
                         // Event params: bytes32 answer, bytes32 indexed question_id, bytes32 history_hash, address indexed user, uint256 bond, uint256 ts, bool is_commitment
-                        const [answerer] = defaultAbiCoder.decode(
-                            ["address"],
-                            log.topics[2]
+                        const [answerer] = decodeAbiParameters(
+                            [{ type: "address", name: "answerer" }],
+                            log.topics[2] as `0x${string}`
                         );
                         const [answer, hash, bond, timestamp] =
-                            defaultAbiCoder.decode(
+                            decodeAbiParameters(
                                 [
-                                    "bytes32",
-                                    "bytes32",
-                                    "uint256",
-                                    "uint256",
-                                    "bool",
+                                    { type: "bytes32", name: "answer" },
+                                    { type: "bytes32", name: "hash" },
+                                    { type: "uint256", name: "bond" },
+                                    { type: "uint256", name: "timestamp" },
+                                    { type: "bool", name: "commitment" },
                                 ],
-                                log.data
+                                log.data as `0x${string}`
```

Comment:
Isn't `log.data` a `Hex`?

Reply:
Fixed in f44ffd9afee046bbfeeca0ecb66ab8586fee4e83.